src/v2/Apps/Conversation/Components/ConversationMessages.tsx

```diff
@@ -0,0 +1,83 @@
+import { groupMessages } from "../Utils/groupMessages"
+import { TimeSince, fromToday } from "./TimeSince"
+import { MessageFragmentContainer as Message } from "./Message"
+import React from "react"
+import { Spacer } from "@artsy/palette"
+import { graphql } from "relay-runtime"
+import { createFragmentContainer } from "react-relay"
+import { ConversationMessages_messages } from "v2/__generated__/ConversationMessages_messages.graphql"
+
+interface ConversationMessageProps {
+  messages: ConversationMessages_messages
+}
+
+export const ConversationMessages = ({
```

Comment:
To make the message logic easier to test I pulled the messages out into their own component. This lets us think about the message UI without having to consider the implications of pagination. It also has the added perk of making the `Conversation` component easier to grok.